### PR TITLE
Changing handling of new peer

### LINKED ISSUE #357: Handle new server from peerbook
When Terminal7 receives and unverified peer it  will pop TWR and ask the user if he wants to authorize it.

```
- not now
- approve
- don't ask me again
```

When the users chooses to approve the client sends a POST to /authorize using with a decrypted fp (using the SSH id keys).

### PR DESCRIPTION
**Pull request by eyal-sasson**
Resolves #357.
Not sure if this was the intention but that's what I understood.

### CODE REVIEW

**Comment by daonb** [comment]
Yesterday I pushed a big change to subs including moving the verifyFP to peerbook.ts so you'll have to pull and refactor.
Some of the logic should stay in shell.ts as verifyFP is used both by "subscribe" and "install" commands where we need minimal user interaction.

File: src/shell.ts [hunk 1]
```diff
@@ -733,6 +733,36 @@ export class Shell {
...
+        this.t.writeln(`Peer ${g.name} is unverified`)
+        const { value } = await Preferences.get({ key: "alwaysVerify" })
+        if (value == "true") {
+            await this.verifyFP(g.fp)
+            return
+        }
+        const fields = [
+            { prompt: "Verify peer" },
+            { prompt: "Not now" },
+            { prompt: "Don't ask again"}
```
Comment (daonb) [inline src/shell.ts:748]: A better prompt will be "Don't ask again and always verify"

File: src/peerbook.ts [hunk 2]
```diff
@@ -124,6 +124,7 @@ export class PeerbookConnection {
                 gate.nameE = terminal7.map.add(gate)
                 gate.open(terminal7.e)
                 ret.push(gate)
+                terminal7.map.shell.verifyPeer(gate).catch()
```
Comment (daonb) [inline src/peerbook.ts:127]: I don't think the `catch` at the end does anything

**Review by daonb — CHANGES_REQUESTED** [review]
Please try and add a test for this in aatp/peerbook_webrtc/admin.spec.ts .

And make sure to pull the latest subs, I've pushed a lot of changes.

File: src/shell.ts (hunk 1 above)
Comment (eyal-sasson) [reply src/shell.ts:597]: I think just "Always verify" works better.

File: src/shell.ts [hunk 3]
```diff
@@ -594,23 +596,19 @@ export class Shell {
         const fields = [
             { prompt: "Verify peer" },
             { prompt: "Not now" },
-            { prompt: "Don't ask again"}
+            { prompt: "Always verify"}
```
Comment (daonb) [inline src/shell.ts:599]: It's better but still a bit weird. Let's remove this option for now. 

Please open an issue with a new feature targeted at those choosing "not now" and giving them a way to verify the peer.

Comment (eyal-sasson) [reply src/shell.ts:599]: How would they verify?

File: src/shell.ts [hunk 4]
```diff
@@ -537,8 +538,9 @@ export class Shell {
                 this.t.writeln("Invalid OTP, please try again")
         }
     }
-    async verifyFP(fp: string, prompt: string) {
-        let validated = false
+    async verifyFP(fp: string, prompt?: string) {
+        return terminal7.pb.verifyFP(fp, prompt)
```
Comment (daonb) [inline src/shell.ts:542]: This function does nothing but calls terminal7.pb.verifyFP, better to remove it and call directly

**Review by daonb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**